lib/Db/MessageMapper.php

```diff
@@ -924,25 +955,30 @@ public function findRelatedData(array $messages, string $userId): array {
 	 * @return int[]
 	 */
 	public function findNewIds(Mailbox $mailbox, array $ids): array {
-		$qb = $this->db->getQueryBuilder();
 		$sub = $this->db->getQueryBuilder();
+		$qb = $this->db->getQueryBuilder();
 
 		$subSelect = $sub
 			->select($sub->func()->max('uid'))
 			->from($this->getTableName())
 			->where(
 				$sub->expr()->eq('mailbox_id', $qb->createNamedParameter($mailbox->getId(), IQueryBuilder::PARAM_INT)),
-				$sub->expr()->in('id', $qb->createNamedParameter($ids, IQueryBuilder::PARAM_INT_ARRAY), IQueryBuilder::PARAM_INT_ARRAY)
+				$sub->expr()->in('id', $qb->createParameter('ids'))
 			);
-		$select = $qb
-			->select('id')
+
+		$qb->select('id')
 			->from($this->getTableName())
 			->where(
-				$qb->expr()->eq('mailbox_id', $qb->createNamedParameter($mailbox->getId(), IQueryBuilder::PARAM_INT)),
-				$qb->expr()->gt('uid', $qb->createFunction('(' . $subSelect->getSQL() . ')'), IQueryBuilder::PARAM_INT)
+				$qb->expr()->eq('mailbox_id', $qb->createNamedParameter($mailbox->getId(), IQueryBuilder::PARAM_INT))
```

Comment:
**Just an idea - feel free to ignore ;)**

We are using the mailbox_id in the outer query and subselect. The current code would pass three parameters to sql server: mailbox_id (outer), mailbox_id (subselect), ids

We have to ways to tell the query builder that's the same parameter: 

1. Define a placeholder `$qb->createNamedParameter($mailbox->getId(), IQueryBuilder::PARAM_INT, ':mailboxId')`
2. Define `$mailboxIdParameter = $qb->createNamedParameter($mailbox->getId(), IQueryBuilder::PARAM_INT)` and use it in both queries.

Reply:
Sure, but what's the benefit?

Reply:
Cleaner code and not sending the same value twice ;)